### PR TITLE
Version util

### PR DESCRIPTION
**Pull request by Justinluu2311**
Library has a version.txt which is used when it cannot pull the latest tag from git. Latest tag from git (if fetched) will be written to version.txt to ensure it is always on the latest version. When pushing, a workflow ensures that the tag automatically has the newest version on github.

Has API endpoint so the app-service can call for its version and the frontend can display it.

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# from https://github.com/anothrNick/github-tag-action, works for merge requests to main
```
Comment (n4dinuh) [inline .github/workflows/workflow.yml]: It may be more informative to name this file `release.yml` rather than `workflow.yml` :)

File: README.md [hunk 2]
```diff
@@ -1,2 +1,10 @@
 # lib-version
 A version-aware library consumed by the URL fishing detection application service
+
+Uses semantic versioning (vMAJOR.MINOR.PATCH)
+
+This library has an endpoint /lib-version for GET requests which will return the library's version.
```
Comment (n4dinuh) [inline README.md]: Bit nitpicky but perhaps use backticks: `/lib-version` for better readability

File: README.md [hunk 3]
```diff
@@ -1,2 +1,10 @@
 # lib-version
 A version-aware library consumed by the URL fishing detection application service
+
+Uses semantic versioning (vMAJOR.MINOR.PATCH)
+
+This library has an endpoint /lib-version for GET requests which will return the library's version. 
+The version comes from either the latest tag on git or the version.txt, depending on which one is available.
```
Comment (n4dinuh) [inline README.md]: Same here: `version.txt`

File: src/VersionUtil.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+            latest_version = max(tag["name"] for tag in tags)
+            with open('../version.txt', 'w') as file:
+                file.write(latest_version)
+            return jsonify("lib-version: " + latest_version)
+        except requests.exceptions.RequestException:  # Handles all requests-related exceptions
+            try:
+                with open("../version.txt", "r") as file:
+                    last_saved_version = file.read().strip()
+                    return jsonify("last saved lib-version: " + last_saved_version)
+            except FileNotFoundError:
```
Comment (n4dinuh) [inline src/VersionUtil.py]: Great that you do proper exception handling!

File: README.md [hunk 5]
```diff
@@ -1,2 +1,10 @@
 # lib-version
 A version-aware library consumed by the URL fishing detection application service
+
+Uses semantic versioning (vMAJOR.MINOR.PATCH)
```
Comment (n4dinuh) [inline README.md]: Can you maybe elaborate on how the semantic versioning works under the hood? On https://github.com/anothrNick/github-tag-action I read stuff about "manual" and "automatic" bumping

File: .github/workflows/workflow.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    permissions:
+      contents: write
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request.merge_commit_sha }}
+        fetch-depth: '0'
+
+    - name: Bump version and push tag
+      uses: anothrNick/github-tag-action@v1 # Don't use @master or @v1 unless you're happy to test the latest version
```
Comment (n4dinuh) [inline .github/workflows/workflow.yml]: Are you sure this action is releasing the library to a Python package registry (e.g. PyPi)? From what I read on https://github.com/anothrNick/github-tag-action this only tags and does a pre-release

Comment (n4dinuh) [reply .github/workflows/workflow.yml]: You can have a look at this workflow in `lib-ml`: https://github.com/remla2024-team14/lib-ml/blob/main/.github/workflows/release.yml

Comment (n4dinuh) [reply .github/workflows/workflow.yml]: This is an example of pushing to GHs _container_ registry: https://github.com/remla2024-team14/model-service/blob/build-release-and-version/.github/workflows/release.yml

File: README.md (hunk 5 above)
Comment (n4dinuh) [reply README.md]: Note that there are also more explicit ways of semantic versioning (https://medium.com/@AranT/auto-tagging-and-using-semantic-versioning-with-github-actions-e40188d12cf4)

**Review by n4dinuh — CHANGES_REQUESTED** [review]

File: .github/workflows/workflow.yml (hunk 6 above)
Comment (Justinluu2311) [reply .github/workflows/workflow.yml]: It only does tags, it doesn't release to pypi, but I'll make it release to pypi

**Comment by n4dinuh** [comment]
Oh and could you maybe extend the README with more info? e.g.
- Explain that this library does not really have a purpose, but was just created for the REMLA course as toy example of version-aware lib
- How to ask for the library's version?

**Review by n4dinuh — APPROVED** [review]
Nice that you got the releasing working!

**Pull request merged**